typing_extensions/src_py3/test_typing_extensions.py

```diff
@@ -0,0 +1,636 @@
+
+        class User: pass
+        class BasicUser(User): pass
+        class ProUser(User): pass
+
+        def new_user(user_class: Type[User]) -> User:
+            return user_class()
+
+        new_user(BasicUser)
+
+    def test_type_typevar(self):
+
+        class User: pass
+        class BasicUser(User): pass
+        class ProUser(User): pass
+
+        U = TypeVar('U', bound=User)
+
+        def new_user(user_class: Type[U]) -> U:
+            return user_class()
+
+        new_user(BasicUser)
+
+    @skipUnless(sys.version_info[:3] != (3, 5, 2),
+                'Python 3.5.2 has a somewhat buggy Type impl')
+    def test_type_optional(self):
+        A = Optional[Type[BaseException]]
+
+        def foo(a: A) -> Optional[BaseException]:
+            if a is None:
+                return None
+            else:
+                return a()
+
+        assert isinstance(foo(KeyboardInterrupt), KeyboardInterrupt)
+        assert foo(None) is None
+
+
+class NewTypeTests(BaseTestCase):
+
+    def test_basic(self):
+        UserId = NewType('UserId', int)
+        UserName = NewType('UserName', str)
+        self.assertIsInstance(UserId(5), int)
+        self.assertIsInstance(UserName('Joe'), str)
+        self.assertEqual(UserId(5) + 1, 6)
+
+    def test_errors(self):
+        UserId = NewType('UserId', int)
+        UserName = NewType('UserName', str)
+        with self.assertRaises(TypeError):
+            issubclass(UserId, int)
+        with self.assertRaises(TypeError):
+            class D(UserName):
+                pass
+
+
+class AllTests(BaseTestCase):
+    def test_typing_extensions_includes_standard(self):
+        a = typing_extensions.__all__
+        self.assertIn('ClassVar', a)
+        self.assertIn('Type', a)
+        self.assertIn('ChainMap', a)
+        self.assertIn('ContextManager', a)
+        self.assertIn('Counter', a)
+        self.assertIn('DefaultDict', a)
+        self.assertIn('Deque', a)
+        self.assertIn('NewType', a)
+        self.assertIn('overload', a)
+        self.assertIn('Text', a)
+        self.assertIn('TYPE_CHECKING', a)
+
+        if ASYNCIO:
+            self.assertIn('Awaitable', a)
+            self.assertIn('AsyncIterator', a)
+            self.assertIn('AsyncIterable', a)
+            self.assertIn('Coroutine', a)
+            self.assertIn('AsyncContextManager', a)
+
+        if PY36:
+            self.assertIn('AsyncGenerator', a)
+        if hasattr(collections_abc, 'Collection'):
+            self.assertIn('Collection', a)
+
+    def test_typing_extensions_defers_when_possible(self):
+        exclude = {'overload', 'Text', 'TYPE_CHECKING'}
+        for item in typing_extensions.__all__:
+            if item not in exclude and hasattr(typing, item):
+                self.assertIs(
+                    getattr(typing_extensions, item),
+                    getattr(typing, item))
+
+
+if __name__ == '__main__':
+   # main(argv=[sys.argv[0]] + sys.argv[2:])
```

Comment:
This was a holdover from the code I previously had for running tests, which we removed. I removed the comment here + in the Python 2 tests.